### ISSUE TITLE
Impossible to login on Nextcloud

### ISSUE THREAD

**Issue by davidv92** [nextcloud/circles#85]
After a restart of the Docker container running Nextcloud 12, I'm not able to login on Nextcloud.

![nc](https://user-images.githubusercontent.com/29569906/27329387-b4171cd8-55b5-11e7-8c73-ff2030b5e43d.png)

 I have theses errors on nextcloud log file : 
```
503 Error: Class 'OCA\Circles\ShareByCircleProvider' not found
#0 [internal function]: {closure}(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))
#1 /nextcloud/3rdparty/sabre/event/lib/EventEmitterTrait.php(105): call_user_func_array(Object(Closure), Array)
#2 /nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php(466): Sabre\Event\EventEmitter->emit('beforeMethod', Array)
#3 /nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php(254): Sabre\DAV\Server->invokeMethod(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))
#4 /nextcloud/remote.php(70): Sabre\DAV\Server->exec()
#5 /nextcloud/remote.php(167): handleException(Object(Error))
#6 {main}

Class 'OCA\Circles\ShareByCircleProvider' not found
#0 /nextcloud/lib/private/Share20/ProviderFactory.php(241): OC\Share20\ProviderFactory->getShareByCircleProvider()
#1 /nextcloud/lib/private/Share20/Manager.php(997): OC\Share20\ProviderFactory->getProviderForType(7)
#2 /nextcloud/apps/files_sharing/lib/MountProvider.php(74): OC\Share20\Manager->getSharedWith('dvalin', 7, NULL, -1)
#3 /nextcloud/lib/private/Files/Config/MountProviderCollection.php(108): OCA\Files_Sharing\MountProvider->getMountsForUser(Object(OC\User\User), Object(OC\Files\Storage\StorageFactory))
#4 /nextcloud/lib/private/Files/Filesystem.php(445): OC\Files\Config\MountProviderCollection->addMountForUser(Object(OC\User\User), Object(OC\Files\Mount\Manager))
#5 /nextcloud/lib/private/Cache/File.php(53): OC\Files\Filesystem::initMountPoints('dvalin')
#6 /nextcloud/lib/private/Cache/File.php(174): OC\Cache\File->getStorage()
#7 /nextcloud/lib/base.php(829): OC\Cache\File->gc()
#8 [internal function]: OC::{closure}(Object(OC\User\User), '')
#9 /nextcloud/lib/private/Hooks/EmitterTrait.php(99): call_user_func_array(Object(Closure), Array)
#10 /nextcloud/lib/private/Hooks/PublicEmitter.php(33): OC\Hooks\BasicEmitter->emit('\\OC\\User', 'postLogin', Array)
#11 /nextcloud/lib/private/User/Session.php(359): OC\Hooks\PublicEmitter->emit('\\OC\\User', 'postLogin', Array)
#12 /nextcloud/core/Controller/LoginController.php(257): OC\User\Session->completeLogin(*** sensitive parameters replaced ***)
#13 [internal function]: OC\Core\Controller\LoginController->tryLogin(*** sensitive parameters replaced ***)
#14 /nextcloud/lib/private/AppFramework/Http/Dispatcher.php(160): call_user_func_array(Array, Array)
#15 /nextcloud/lib/private/AppFramework/Http/Dispatcher.php(90): OC\AppFramework\Http\Dispatcher->executeController(Object(OC\Core\Controller\LoginController), 'tryLogin')
#16 /nextcloud/lib/private/AppFramework/App.php(114): OC\AppFramework\Http\Dispatcher->dispatch(Object(OC\Core\Controller\LoginController), 'tryLogin')
#17 /nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php(47): OC\AppFramework\App::main('OC\\Core\\Control...', 'tryLogin', Object(OC\AppFramework\DependencyInjection\DIContainer), Array)
#18 [internal function]: OC\AppFramework\Routing\RouteActionHandler->__invoke(Array)
#19 /nextcloud/lib/private/Route/Router.php(299): call_user_func(Object(OC\AppFramework\Routing\RouteActionHandler), Array)
#20 /nextcloud/lib/base.php(1000): OC\Route\Router->match('/login')
#21 /nextcloud/index.php(40): OC::handleRequest()
#22 {main}

HTTP/1.1 503 Error: Class 'OCA\Circles\ShareByCircleProvider' not found
#0 [internal function]: {closure}(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))
#1 /nextcloud/3rdparty/sabre/event/lib/EventEmitterTrait.php(105): call_user_func_array(Object(Closure), Array)
#2 /nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php(466): Sabre\Event\EventEmitter->emit('beforeMethod', Array)
#3 /nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php(254): Sabre\DAV\Server->invokeMethod(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))
#4 /nextcloud/remote.php(70): Sabre\DAV\Server->exec()
#5 /nextcloud/remote.php(167): handleException(Object(Error))
#6 {main}
```

Before the restart of the Docker container "Circles" app was activated.
Currently "Circles" app is not installed because by default after a restart the Docker container don't include this application.

**Comment by daita** [comment]
i will have a look, in the meanwhile you should enable circles from the command line: ./occ app:enable circles

**Comment by davidv92** [comment]
Thank you for you quick feedback.
I can't enable the app with `occ` because the app is not installed.
I can't install the app throw the web interface, so I will try to install it manually and then do `occ app:enable circles`.

**Comment by daita** [comment]
PR have been merged on nextcloud and will be available on 12.0.1
